lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -285,7 +285,12 @@ def add_stored_credential_using_normalized_fields(xml, options)
         if options[:stored_credential][:initial_transaction]
           xml.tag! 'storedCredentials', 'usage' => 'FIRST'
         else
-          reason = options[:stored_credential][:recurring] ? 'RECURRING' : 'UNSCHEDULED'
+          reason = case options[:stored_credential][:transaction_type]
```

Comment:
I don't know if I'd use the term "transaction type" for the normalization, seems a little general for this usage. May I suggest "schedule"?

Reply:
I agree, `transaction_type` is pretty generic. I looked up a couple of gateways docs and they all had a different term for this 😸The ones I saw were:

* `reason`
* `type`
* `sequence`
* `indicator`

Of those, I think the word `reason` is pretty good. As in, "the _reason_ for this transaction is a recurring monthly subscription". Or going along those same lines the word `purpose` is another good synonym. 

I do think `schedule` still feels a bit vague since I would then assume it's a recurring transaction and the value would be something along the lines of "monthly".

Reply:
I agree `transaction type` is too general in this context - and could be confused with the general transaction type (auth, credit, etc). 

I like `reason`, though I wonder if it would be interpreted as a free form text field versus a using `*_type` as the field name that seems to imply a limited set of types.

Reply:
What values do we expect this field to hold?

Reply:
> I wonder if it would be interpreted as a free form text field versus a using `*_type` as the field name that seems to imply a limited set of types.

🤔 ya that's a good point.

> What values do we expect this field to hold?

`recurring`, `unscheduled`, `installment`. Maybe we can call it `reason_type`? 😸

Reply:
Digging into synonyms, do any of these feel more appropriate and encompassing?

* frequency
* cycle
* period

Reply:
I'm cool with `reason_type`

Reply:
@jknipp my only hesitation with those terms is that they imply either that it is an ongoing thing- which it may not be- or that there is a set time frame between transactions.

I think I'm going to go with `reason_type` which hopefully conveys the "reason" for a transaction (i.e. this transaction is happening cause it's a recurring transaction or this transaction is happening because a customer did a one off purchase on our website) with a `_type` suffix to imply there's a limited number of reasons.

Reply:
👍 Works for me.